### PR TITLE
Hill cipher Explaination

### PR DESCRIPTION
**Pull request by Jenil-S**

### CODE REVIEW

**Comment by Jenil-S** [comment]
@Panquesito7 Plz review it

**Comment by Jenil-S** [comment]
@cclauss just review it

File: en/Ciphers/hill_cipher.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+### Encryption
+
+1. We have to write key as an n × n matrix as 
+ ```
+    [6 24 1]
+    [13 16 10]
+    [20 17 15]
+```
+2. Same way convert PT into vector as
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:21]:
```suggestion
2. Same way convert PT into a vector as

```

File: en/Ciphers/hill_cipher.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+``` 
+    [0]
+    [2]
+    [19]
+```
+3. Now we need to encipher the vector by just multiplying these two matrices
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:27]:
```suggestion
3. Now, we need to encipher the vector by just multiplying these two matrices

```

File: en/Ciphers/hill_cipher.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    [2]
+    [19]
+```
+3. Now we need to encipher the vector by just multiplying these two matrices 
+```
+    [6 24 1]        [0]         [67]        [15]
+    [13 16 10]  *   [2]     =   [222]   ≈   [4]   (mod 26)
+    [20 17 15]      [19]        [319]       [7]
+```
+So we will get the encryted text as **POH**
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:33]:
```suggestion
So we will get the encrypted text as **POH**.

```

File: en/Ciphers/hill_cipher.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ ### Decryption
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:34]:
```suggestion
### Decryption

```

File: en/Ciphers/hill_cipher.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ 1. We need to first inverse our key matrix
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:35]:
```suggestion
1. We need to first inverse our key matrix
 
```

File: en/Ciphers/hill_cipher.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+So we will get the encryted text as **POH**
+ ### Decryption
+ 1. We need to first inverse our key matrix 
+ ```
+            -1 
+    [6 24 1]        [8 5 10]
+    [13 16 10]   ≈  [21 8 21]  (mod 26) 
+    [20 17 15]      [21 12 8]
+ ```
+2. For prevoius Cipher Text **POH**
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:42]:
```suggestion
2. For the previous Cipher Text **POH**

```

File: en/Ciphers/hill_cipher.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+```
+    [8 5 10]      [15]      [260]     [0]
+    [21 8 21]  *  [14]  ≈   [574]  ≈  [2]    (mod 26)  ≈ ACT
+    [21 12 8]     [7]       [539]     [19]
+```
+## Implementation
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:48]:
```suggestion
## Implementation

```

File: en/Ciphers/hill_cipher.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+[**Python**](https://github.com/TheAlgorithms/Python/blob/master/ciphers/hill_cipher.py)
+
+## Video Explanation
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:51]:
```suggestion
## Video Explanation

```

File: en/Ciphers/hill_cipher.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+[A video explanation of hill cipher](https://www.youtube.com/watch?v=6T46sgty4Mk)
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:52]:
```suggestion
[A video explanation of the Hill Cipher](https://www.youtube.com/watch?v=6T46sgty4Mk)
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
👍 Good work.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please accept my suggestions mentioned above.

**Comment by Jenil-S** [comment]
> Please accept my suggestions mentioned above.

I had done all the required changes

**Comment by Jenil-S** [comment]
> Please accept my suggestions mentioned above.

what else to do??

**Comment by Jenil-S** [comment]
> Please accept my suggestions mentioned above.

I think now its done you can merge it..

File: en/Ciphers/hill_cipher.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
+# Hill Cipher
+
+As it was invented by Lester.S.Hill,this algorithm was named hill cipher.
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:3]:
```suggestion
As it was invented by Lester.S.Hill, this algorithm was named Hill Cipher.
```

File: en/Ciphers/hill_cipher.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+Hill cipher is a polygraphic substitution cipher based on linear algebra. Each letter is represented by a number modulo 26. Often the simple scheme `A = 0, B = 1, …, Z = 25` is used, but this is not an essential feature of the cipher. To encrypt a message, each block of `n` letters (considered as an n-component vector) is multiplied by an invertible `n × n` matrix, against modulus 26. To decrypt the message, each block is multiplied by the inverse of the matrix used for encryption.
+
+## Example
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:7]:
```suggestion
## Example

```

File: en/Ciphers/hill_cipher.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+Suppose we take an example as:
+Plain Text (PT):ACT
+key:`GYBNQKURP`
+## Steps
+
+### Encryption
+
+1. We have to write key as an `n × n` matrix as
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:15]:
```suggestion
1. We have to write key as an `n × n` matrix as

```

File: en/Ciphers/hill_cipher.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+### Encryption
+
+1. We have to write key as an `n × n` matrix as 
+ ```
+    [6 24 1]
+    [13 16 10]
+    [20 17 15]
+```
+2. Same way convert PT into a vector as
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:21]:
```suggestion
2. Same way convert PT into a vector as

```

File: en/Ciphers/hill_cipher.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+``` 
+    [0]
+    [2]
+    [19]
+```
+3. Now, we need to encipher the vector by just multiplying these two matrices
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:27]:
```suggestion
3. Now, we need to encipher the vector by just multiplying these two matrices

```

File: en/Ciphers/hill_cipher.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+```
+3. Now, we need to encipher the vector by just multiplying these two matrices 
+```
+    [6 24 1]        [0]         [67]        [15]
+    [13 16 10]  *   [2]     =   [222]   ≈   [4]   (mod 26)
+    [20 17 15]      [19]        [319]       [7]
+```
+So we will get the encrypted text as **POH**.
+
+### Decryption
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:35]:
```suggestion
### Decryption

```

File: en/Ciphers/hill_cipher.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+1. We need to first inverse our key matrix 
+ ```
+            -1 
+    [6 24 1]        [8 5 10]
+    [13 16 10]   ≈  [21 8 21]  (mod 26) 
+    [20 17 15]      [21 12 8]
+ ```
+2. For the prevoius Cipher Text **POH**
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:43]:
```suggestion
2. For the previous Cipher Text **POH**

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: en/Ciphers/hill_cipher.md [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+
+As it was invented by Lester.S.Hill, this algorithm was named Hill Cipher.
+
+Hill cipher is a polygraphic substitution cipher based on linear algebra. Each letter is represented by a number modulo 26. Often the simple scheme `A = 0, B = 1, …, Z = 25` is used, but this is not an essential feature of the cipher. To encrypt a message, each block of `n` letters (considered as an n-component vector) is multiplied by an invertible `n × n` matrix, against modulus 26. To decrypt the message, each block is multiplied by the inverse of the matrix used for encryption.
+
+## Example
+
+Suppose we take an example as:
+Plain Text (PT):ACT
+key:`GYBNQKURP`
```
Comment (Panquesito7) [inline en/Ciphers/hill_cipher.md:11]:
```suggestion
key:`GYBNQKURP`

```

**Review by Panquesito7 — DISMISSED** [review]
Great work, @Jenil-S. LGTM. Thank you for your contribution! 👍 🎉

**Review by appgurueu — DISMISSED** [review]
Explanation is fine

**Review by cclauss — APPROVED** [review]

**Pull request merged**